Sources/LibAuk/Storage/SecureStorage.swift

```diff
@@ -40,19 +38,125 @@ public protocol SecureStorageProtocol {
     func exportSeed() -> AnyPublisher<Seed, Error>
     func exportMnemonicWords() -> AnyPublisher<[String], Error>
     func removeKeys() -> AnyPublisher<Void, Error>
+    func setSeed(seed: Seed, isPrivate: Bool) -> AnyPublisher<Bool, Error>
+    func removeSeed() -> AnyPublisher<Bool, Error>
+    func generateSeedPublicData(seed: Seed) throws -> SeedPublicData?
+    func setData(_ data: Data, forKey: String, isSync: Bool, isPrivate: Bool) -> Bool
 }
 
 class SecureStorage: SecureStorageProtocol {
-    
+
     private let keychain: KeychainProtocol
     
+    private let preGenerateAddressLimit: Int = 100
+
     init(keychain: KeychainProtocol = Keychain()) {
         self.keychain = keychain
     }
     
-    func createKey(passphrase: String? = "", name: String) -> AnyPublisher<Void, Error> {
+    internal func generateEthAddresses(mnemonic: BIP39Mnemonic, passphrase: String?, start: Int = 0, end: Int = 100) -> [Int: String] {
+        var ethAddresses: [Int: String] = [:]
+
+        for index in start...end {
+            do {
+                // Generate Ethereum private key for the current index
+                let privateKey = try Keys.ethereumPrivateKeyWithIndex(mnemonic: mnemonic, passphrase: passphrase, index: index)
+
+                // Derive Ethereum address from the private key
+                let address = privateKey.address.hex(eip55: true)
+
+                // Store the address in the dictionary
+                ethAddresses[index] = address
+            } catch {
+                // Handle errors if private key generation fails
+                print("Error generating private key for index \(index): \(error)")
```

Comment:
if there's 1 failure generating the address, the index of the following addresses will become wrong?